### PR TITLE
UAR 325 Payment failed journey for Update

### PR DESCRIPTION
**Pull request by keelan-scott**
### JIRA link
https://companieshouse.atlassian.net/jira/software/c/projects/UAR/boards/481?modal=detail&selectedIssue=UAR-325&sprint=3028

### Change description

Redirects to payment failed page in update with appropriate header for this journey. Header changes depending what journey user is going through - update or register.

### Work checklist

- [ ] Tests added where applicable
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/routes/index.ts [hunk 1]
```diff
@@ -386,6 +387,8 @@ router.route(config.UPDATE_BENEFICIAL_OWNER_STATEMENTS_URL)
 
 router.get(config.OVERSEAS_ENTITY_PAYMENT_WITH_TRANSACTION_URL, authentication, companyAuthentication, overseasEntityPayment.get);
 
+router.get(config.UPDATE_PAYMENT_FAILED_URL, authentication, updatePaymentFailed.get);
```
Comment (AndreKramer7654) [inline src/routes/index.ts:390]: need companyAuthentication after authentication,

Comment (AndreKramer7654) [reply src/routes/index.ts:390]: which would need testing thoroughly. I'm assuming the user is authenticated for the company when the re-direct happens. But may be more important to show error page in all cases. Not sure now.

Comment (AndreKramer7654) [reply src/routes/index.ts:390]: Better to show failure.

File: views/includes/page-templates/page-header.html [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+{% block beforeContent %}
+  {% if isUpdate%}
+    {% include "../update_phase_banner.html" %}
+  {% else %}
+    {% include "../phase_banner.html" %}
+  {% endif %}
+  {% include "../no-back-link.html" %}
+  {% include "../sign-out-user-banner.html" %}
+{% endblock %}
```
Comment (corybudd-ch) [inline views/includes/page-templates/page-header.html:21]: just a question - was it a requirement to add this beforeContent block or what does it do? Before your changes the `views/payment-failed.html` view didn't have this.

Comment (corybudd-ch) [reply views/includes/page-templates/page-header.html:11]: sorry to be even more annoying now but does removing that code now get rid of the need for this new file? Would sonarqube complain about duplication anymore?

Comment (joshuahigginsonCH) [reply views/includes/page-templates/page-header.html:11]: i'd asked this same question but wasn't sure, might be worth a push to check and revert if it complains

Comment (keelan-scott) [reply views/includes/page-templates/page-header.html:11]: I'll check and see.

Comment (corybudd-ch) [reply views/includes/page-templates/page-header.html]: Still complaining?? Ugh. Stupid Sonar. Just rename `titleTemplate` to something else, `pageTitle` maybe?

**Review by corybudd-ch — APPROVED** [review]

**Review by joshuahigginsonCH — APPROVED** [review]

**Pull request merged**